cosmos/scheduler/cmd/scheduler.go

```diff
@@ -0,0 +1,247 @@
+package scheduler
+
+import (
+	"context"
+	"log"
+
+	"github.com/mongodb/mongo-go-driver/bson"
+	"github.com/mongodb/mongo-go-driver/mongo"
+	"github.com/scality/zenko/cosmos/api/types/v1alpha1"
+	clientV1alpha1 "github.com/scality/zenko/cosmos/clientset/v1alpha1"
+	"github.com/scality/zenko/cosmos/scheduler/pkg"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Scheduler represents a Cosmos Scheduler instance
+type Scheduler struct {
+	KubeAlpha     *clientV1alpha1.ExampleV1Alpha1Client
+	KubeClientset *kubernetes.Clientset
+	Pensieve      *pensieve.Helper
+	Namespace     string
+	Cloudserver   string
+	MongodbClient *mongo.Client
+}
+
+// Run starts the Cosmos Scheduler
+func (s *Scheduler) Run() {
+	log.Println("starting cosmos scheduler")
+	exists, err := s.getIngestionSecret()
+	if err != nil {
+		panic(err.Error())
+	}
+	if !exists {
+		if err = s.createIngestionSecret(); err != nil {
+			panic(err.Error())
+		}
+	}
+	msgCh := make(chan interface{}, 1)
+	go func() {
+		for {
+			s.handler(msgCh)
```

Comment:
for clarification, why does the `handler` need to quit for each "overlay update"? I may be missing something, but it doesn't look like the `handler` does anything with that event. For each `__metadata` changes, a check for the latest overlay configuration is always performed, so the `handler` routine doesn't need to "restart"

Reply:
The reason for that is because every time the handler restarts, it starts a new `s.watchBucketEvents()` which will create a new watch stream with a new set of buckets.